puppet/lib/puppet/util/puppetdb/http.rb

```diff
@@ -86,14 +86,21 @@ def self.action(path_suffix, &http_callback)
         end
       end
 
-      if response.nil?
-
+      if response.nil? or response == :notfound
         if server_url_config
           server_url_strings = Puppet::Util::Puppetdb.config.server_urls.map {|url| url.to_s}.join(', ')
-          raise Puppet::Error, "Failed to execute '#{path_suffix}' on any of the following 'server_urls': #{server_url_strings}"
+          if response == :notfound
+            raise NotFoundError, "Failed to find '#{path_suffix}' on any of the following 'server_urls': #{server_url_strings}"
```

Comment:
We raise a not found instead of just a puppet::error here, so in the indirector it can be caught, and we can return an empty value - as per the inventory service API.